docs/topics/azure-stack.md

```diff
@@ -0,0 +1,154 @@
+
+### masterProfile
+
+`masterProfile` describes the settings for master configuration.
+
+| Name                            | Required | Description|
+| ------------------------------- | -------- | ---------- |
+| vmsize                          | yes      | Specifies a valid [Azure Stack VM size](https://docs.microsoft.com/en-us/azure-stack/user/azure-stack-vm-sizes). |
+| distro                          | yes      | Specifies the masters' Linux distribution. Currently supported values are: `"ubuntu"` and `"aks"`. The latter is a custom image based on ubuntu-16.04 that comes with pre-installed software necessary for Kubernetes deployments. `"aks"` should be selected if you target a disconnected Azure Stack instance. |
+
+### agentPoolProfiles
+
+`agentPoolProfiles` are used to create agents with different capabilities.
+
+| Name                            | Required | Description|
+| ------------------------------- | -------- | ---------- |
+| vmsize                          | yes      | Describes a valid [Azure Stack VM size](https://docs.microsoft.com/en-us/azure-stack/user/azure-stack-vm-sizes). |
+| distro                          | yes      | Specifies the masters' Linux distribution. Currently supported values are: `"ubuntu"` and `"aks"`. The latter is a custom image based on ubuntu-16.04 that comes with pre-installed software necessary for Kubernetes deployments. `"aks"` should be selected if you target a disconnected Azure Stack instance. |
+| availabilityProfile             | yes      | Only `"AvailabilitySet"` is currently supported. |
+| acceleratedNetworkingEnabled    | yes      | Use `Azure Accelerated Networking` feature for Linux agents. This property should be always set to `"false"`. |
+
+## Azure Stack Instances Registered with Azure's China cloud
+
+If your Azure Stack instance is located in China, then the `dependenciesLocation` property of your cluster definition should be set to `"china"`. This switch ensures that the provisioning process fetches software dependencies from reachable hosts within China's mainland.
+
+## Disconnected Azure Stack Instances
+
+By default, the AKS Engine provisioning process relies on an internet connection to download the software dependencies required to create or upgrade a cluster (Kubernetes images, etcd binaries, network plugins and so on).
+
+If your Azure Stack instance is air-gapped or if network connectivity in your geographical location is not reliable, then the default approach will not work, take a long time or timeout due to transient networking issues.
+
+With these challenges in mind, you can choose to set the `distro` property of your cluster definition to `"aks"`. This change will instruct AKS Engine to deploy VM nodes using a base OS image call `AKS Base Image`. This custom image already contains the required software dependencies in its file system. Hence, internet connectivity won’t be required during the provisioning process.
```

Comment:
sorry to go back and forth. The distro rename change is already merged in master so I think it's fair to update it here. The documentation will only be in master and will be released at the same time as the distro rename. I was just worried that people would use these instructions with v0.36.2 since it's mentioned at the top of the doc. If that's the case, maybe add a disclaimer like this:
"you can choose to set the `distro` property of your cluster definition to `"aks-ubuntu-16.04"` (or `"aks"` if you are using AKS Engine version v0.36.x or lower)."

what do you think?

Reply:
Because `aks` works with 0.36.2, I was planning to add that clarification after the new distro strings are officially released. It is not critical to have perfect timing at this point.